### ISSUE TITLE
Option to only store quantized vectors on disk

### ISSUE THREAD

**Issue by MaximilianAzendorf** [qdrant/qdrant#2632]
**Is your feature request related to a problem? Please describe.**
I have a use case for qdrant where I want to store a large vector set (billions of vectors) on a disk-space constrained deployment. Accuracy is not a top priority, so I am using scalar quantization.

**Describe the solution you'd like**
As far as I understand, qdrant currently always stores the original float32 vectors alongside the quantized vectors on disk. An option where qdrant discards the original vectors and only stores the quantized vectors on disk would greatly reduce disk usage of the vector collection.

**Describe alternatives you've considered**
No alternatives as far as I am aware of.

**Additional context**
I would be willing to implement this as I have experience in Rust but I am relatively new to qdrant and don't know how to tackle this.

**Comment by generall** [comment]
Hi @MaximilianAzendorf, unfortunately there are a lot of critical functions that depend on having original vectors.
Additionally, the main purpose of quantization was to reduce RAM usage, while disk space was always considered cheap enough to store all vectors.

For that reason we will not implement the option you described.

One possible alternative for you is to build version of Qdrant with an option to use f16 or int8 as a default type for storing vectors, but I don't have an estimation of how hard it might be. (probably for in8 is much harder than f16)

**Comment by the-jb** [comment]
I also agree with @MaximilianAzendorf for supporting billions of vectors.
The disk space should be considered for storing billions of high-dimensional vectors, especially with NVMe.
At least, it would be great if supporting naive bit vector datatype.

**Comment by FaisalAliShah** [comment]
@generall is this still the case?

**Comment by generall** [comment]
uint8 and float16 are supported as a datatype for original vectors. But this can work in combination with quantization as well
